crates/spirv-std/src/lib.rs

```diff
@@ -51,6 +51,26 @@ pub use glam;
 pub use num_traits;
 pub use textures::*;
 
+/// Calls the `OpDemoteToHelperInvocationEXT` instruction, which corresponds to discard() in HLSL
+pub fn demote_to_helper_invocation() {
+    #[cfg(target_arch = "spirv")]
+    unsafe {
+        asm!(
+            "OpExtension \"SPV_EXT_demote_to_helper_invocation\"",
+            "OpCapability DemoteToHelperInvocationEXT",
+            "OpDemoteToHelperInvocationEXT"
```

Comment:
Oh, right, the other reason to have a token is because we could support it in the CPU runner if it just had a `&AtomicBool` (or just `&Cell<bool>`) indicating it's a helper invocation and the output shouldn't be written.

Far less sure about what to do with `OpKill` tho, short of a `panic!` with a special type that can be caught and acted upon. Anyway that's just more tangents.